addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -421,6 +421,210 @@ QUnit.test('sidebar: add channel', async function (assert) {
+        {"id":20,"name":"General","state":"open","channel_type":"channel","public":"public","info":"unsubscribe"}
+    ];
+    await afterNextRender(() =>
+        this.env.services.bus_service.trigger('notification', [notifConfirmUnpin])
+    );
+    assert.containsNone(document.body, '.o_DiscussSidebar_list .o_DiscussSidebarItem', "The channel still is not in the sidebar");
+    assert.strictEqual(
+        document.body.querySelector('.breadcrumb-item.active').textContent,
+        'Inbox',
+        "The inbox is displayed"
+    );
+    /**
+     * Step 2c: Although the bus notifications are sent just after leaving
+     * Some other Tab may send some notification as well
+     * Those asserts are merely a protection
+     */
+    const notifConfirmFold = [
+        ["dbName","res.partner",1],
+        {"id":20,"name":"General","state":"closed", "is_minimized": false, "channel_type":"channel","public":"public"}
+    ];
+    try {
+        await afterNextRender(
+            () => this.env.services.bus_service.trigger('notification', [notifConfirmFold]),
+            100 // we wait a bit, but we know it is gonna fail
+        );
+    } catch (e) {
+        // Normally there is nothing to do since the channel is already closed and destroyed
+        assert.strictEqual(e.message, "Timeout: the render didn't start.");
+    }
+    assert.containsNone(document.body, '.o_DiscussSidebar_list .o_DiscussSidebarItem', "The channel still is not in the sidebar");
+    assert.strictEqual(
+        document.body.querySelector('.breadcrumb-item.active').textContent,
+        'Inbox',
+        "The inbox is displayed"
+    );
+
+    /**
+     * Step 3: Some other tab has opened a chatwindow that displays this channel
+     * so the channel is back to being pinned
+     * but the inbox is still opened
+     */
+    const notifConfirmUnFold = [
+        ["dbName","res.partner",1],
+        {"id":20,"name":"General","state":"open","channel_type":"channel","public":"public"}
+    ];
+    await afterNextRender(() =>
+        this.env.services.bus_service.trigger('notification', [notifConfirmUnFold])
+    );
+    assert.containsOnce(document.body, '.o_DiscussSidebar_list .o_DiscussSidebarItem', "The channel has been brought back in the sidebar");
+    assert.strictEqual(
+        document.body.querySelector('.breadcrumb-item.active').textContent,
+        'Inbox',
+        "The inbox is displayed"
+    );
+});
+
+QUnit.test('sidebar: unpin channel from bus', async function (assert) {
+    assert.expect(6);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+
+    await this.start();
+    assert.strictEqual(
+        document.body.querySelector('.breadcrumb-item.active').textContent,
+        'Inbox',
+        "The inbox is displayed"
+    );
```

Comment:
Using data- is both weak to the localId of the inbox changing, and weak to relying on localId in the first place. That data shouldn't even be in the DOM in the first place if it wasn't for the test currently using it.

It's time we find a proper solution for this, and only thing that makes sense for a functional test is asserting what the user sees. If it's not enough for some reason, then let's just assert the state of models directly instead of propagating any data- in the DOM, even more so a localId.

Reply:
> Using data- is both weak to the localId of the inbox changing, and weak to relying on localId in the first place. That data shouldn't even be in the DOM in the first place if it wasn't for the test currently using it.
> 
> It's time we find a proper solution for this, and only thing that makes sense for a functional test is asserting what the user sees. If it's not enough for some reason, then let's just assert the state of models directly instead of propagating any data- in the DOM, even more so a localId.

If local id of inbox changes, then assertion still works, so I don't see how it's "weak to the localId of the inbox changing". And the argument "weak to relying on localId in the first place" is kinda wrong because that's much more robust than say assertion(s) on name/content/title/number of messages/content of messages/etc, which all are super noisy and are just a bunch of heuristics...

The only better solution I have in mind is to instead define and use `data-thread-id` and `data-thread-model` on thread viewer:

- both DOM data 100% accurately reflect the intent of component to display given thread
- both DOM data do not leak implementation detail like `localId` (here by leak, I mean test is aware that we use local ids to identify threads)

It makes no functional sense to display them in UI, so it's totally legitimate to use `data-` for this kind of information.

As I said, you can see like a meta-data on presentation, like when you look at a painting and on its back there are some data like name of owner or serial number... Which are super handy stuffs for, like, uniquely identify the painting. But you don't want to put that on the painting itself... At least most of the time!

Reply:
Ok for `data-thread-id` & `data-thread-model`, this is already much better than `localId`.

Can we find a naming that convey those data are only a signature, and that their presence or value should be completely ignored by business code?